gef.py

```diff
@@ -8848,6 +8848,60 @@ def do_invoke(self, argv: List[str]) -> None:
             err("No address mapping information found")
             return
 
+        # Parse args:
+        #  - `-a` / `--addr`:
+        #    - filter by address -> parses the next arg as an int or asks gdb
+        #    the value
+        #  - `-n` / `--name`:
+        #    - filter based on section name
+        #  - If nothing is specified, print a warning and guess the type
+        current_type = 0 # 0:None, 1:Address, 2:Name
+
+        addrs = {}
+        names = []
+
+        for arg in argv:
```

Comment:
Was there a reason not to use `@parse_arguments` here?

Reply:
Can we make `parse_arguments` parsing something that could look like `--addr xxx --addr yyy --addr zzz`? If so, there is no valid reason not to use it, I just didn't know this decorator.

Reply:
@hugsy idk if you have the answer, I'm stopping for today but if this is not possible to do then I think we should stick to this logic, otherwise I'll rewrite this part for using `parse_arguments` later

Reply:
> @hugsy idk if you have the answer, I'm stopping for today but if this is not possible to do then I think we should stick to this logic, otherwise I'll rewrite this part for using `parse_arguments` later

I'm not going to be on my PC for a while but a quick grep shows we use list at several locations if you need examples:
https://github.com/hugsy/gef/blob/c450a2d5fdd6f3e58fcadf8fe97365c8a36c6172/gef.py#L7094
https://github.com/hugsy/gef/blob/c450a2d5fdd6f3e58fcadf8fe97365c8a36c6172/gef.py#L8505
https://github.com/hugsy/gef/blob/c450a2d5fdd6f3e58fcadf8fe97365c8a36c6172/gef.py#L5014

Reply:
It is a wrapper on `argparse.ArgumentParser` so anything it supports everything supported by the latter.

Reply:
Also if it wasn't, I feel that it would have been a good thing to add

Reply:
Done, PTAL :)